python/pyarrow/array.pxi

```diff
@@ -2178,7 +2178,10 @@ cdef class StructArray(Array):
 
         arrays = [asarray(x) for x in arrays]
         for arr in arrays:
-            c_arrays.push_back(pyarrow_unwrap_array(arr))
+            c_array = pyarrow_unwrap_array(arr)
+            if c_array == nullptr:
+                raise TypeError(f"Unsupported array type {arr.__class__}")
```

Comment:
To improve this error message, maybe it can something like "Expected Array, got {..} instead) ?

Reply:
👍  done